src/flame/variations/transformFunctions.ts

```diff
@@ -0,0 +1,311 @@
+    let p1 = cos(theta - r);
+    let p03 = p0 * p0 * p0;
+    let p13 = p1 * p1 * p1;
+    return r * vec2f((p03 + p13), (p03 - p13));
+  }`)
+
+const julia = simpleFn(
+  /* wgsl */ `
+  (pos: vec2f) -> vec2f {
+    let sqrtr = sqrt(length(pos));
+    let theta = atan2(pos.y, pos.x);
+    let rand = random();
+    let omega = select(0, PI, random() > 0.5);
+    let angle = theta / 2.0 + omega;
+    return sqrtr * vec2f(cos(angle), sin(angle));
+  }`,
+  { random, PI },
+)
+
+const bent = simpleFn(/* wgsl */ `
+  (pos: vec2f) -> vec2f {
+    let fx = select(pos.x, 2.0 * pos.x, pos.x < 0);
+    let fy = select(pos.y, pos.y / 2.0, pos.y < 0);
+    return vec2f(fx, fy);
+  }`)
+
+const waves = dependentFn(/* wgsl */ `
+  (pos: vec2f, T: AffineParams) -> vec2f {
+    let xSinArg = pos.y / (T.c * T.c); 
+    let ySinArg = pos.x / (T.f * T.f); 
+    return vec2f(
+      pos.x + T.b * sin(xSinArg),
+      pos.y + T.e * sin(ySinArg),
+    );
+  }`)
+
+const fisheye = simpleFn(/* wgsl */ `
+  (pos: vec2f) -> vec2f {
+    let r = length(pos);
+    let factor = 2 / (r + 1); 
+    return factor * pos.yx;
+  }`)
+
+const eyefish = simpleFn(/* wgsl */ `
+  (pos: vec2f) -> vec2f {
+    let r = length(pos);
+    let factor = 2 / (r + 1); 
+    return factor * pos;
+  }`)
+
+const popcorn = dependentFn(/* wgsl */ `
+  (pos: vec2f, T: AffineParams) -> vec2f {
+    return pos + vec2f(
+      T.c * sin(tan(3 * pos.y)),
+      T.f * sin(tan(3 * pos.x)),
+    );
+  }`)
+
+const exponential = simpleFn(
+  /* wgsl */ `
+  (pos: vec2f) -> vec2f {
+    let factor = exp(pos.x - 1);
+    let piY = PI * pos.y;
+    return factor * vec2f(cos(piY), sin(piY));
+  }`,
+  { PI },
+)
+
+const power = simpleFn(/* wgsl */ `
+  (pos: vec2f) -> vec2f {
+    let r = length(pos);
+    let theta = atan2(pos.y, pos.x);
+    let sinTheta = sin(theta);
+    let factor = pow(r, sinTheta);
+    return factor * vec2f(cos(theta), sinTheta);
+  }`)
+
+const cosine = simpleFn(
+  /* wgsl */ `
+  (pos: vec2f) -> vec2f {
+    let piX = PI * pos.x;
+    return vec2f(
+      cos(piX) * cosh(pos.y),
+      -sin(piX) * sinh(pos.y)
+    );
+  }`,
+  { PI },
+)
+
+const rings = dependentFn(/* wgsl */ `
+  (pos: vec2f, T: AffineParams) -> vec2f {
+    let c2 = T.c * T.c;
+    let r = length(pos); 
+    let theta = atan2(pos.y, pos.x);
+    let factor = (r + c2) % (2 * c2) - c2 + r * (1 - c2);
+    return factor * vec2f(cos(theta), sin(theta));
+  }`)
+
+const fan = dependentFn(
+  /* wgsl */ `
+  (pos: vec2f, T: AffineParams) -> vec2f {
+    let t = PI * T.c * T.c;
+    let r = length(pos); 
+    let theta = atan2(pos.y, pos.x);
+
+    let thalf = t / 2;
+    let trueAngle = theta - thalf;
+    let falseAngle = theta + thalf;
+    let modCond = (theta + T.f) % t;
+    let angle = select(falseAngle, trueAngle, modCond > thalf);
+    return r * vec2f(cos(angle), sin(angle));
+  }`,
+  { PI },
+)
+
+export type TransformFunction = keyof typeof transformFunctions
+export const transformFunctions = {
```

Comment:
Instead of exporting like this, I'd export individual items and then combine them in `index.ts`

Reply:
Created index